### PR TITLE
Make sure examples/elf testbuild pass in parallel build

### PR DESCRIPTION
**Pull request by liuguo09**

### CODE REVIEW

File: examples/elf/tests/errno/Makefile [hunk 1]
```diff
@@ -86,6 +86,6 @@ clean:
 	$(call DELFILE, $(BIN))
 	$(call CLEAN)
 
-install:
+install: $(BIN)
```
Comment (yamt) [inline examples/elf/tests/errno/Makefile]: if we do this at this level, we should remove "build" step from examples/elf/tests/Makefile.

an alternative: https://github.com/apache/incubator-nuttx-apps/pull/103

File: examples/elf/Makefile [hunk 2]
```diff
@@ -57,6 +57,8 @@ else
   FSIMG_HDR = $(TESTS_DIR)/cromfs.h
 endif
 
+elf_main.c: tests/symtab.c
```
Comment (yamt) [inline examples/elf/Makefile]: i guess it's more straightforward to require $(FSIMG_HDR) instead.

File: examples/elf/tests/errno/Makefile (hunk 1 above)
Comment (liuguo09) [reply examples/elf/tests/errno/Makefile]: You are right. I think your approach is simple and more general.

File: examples/elf/Makefile (hunk 2 above)
Comment (xiaoxiang781216) [reply examples/elf/Makefile]: But, who do trigger thetest folder build? we have to duplicated the test/symtab.c action here, but this change will trigger test folder build twice.

Comment (yamt) [reply examples/elf/Makefile]: i'm not sure what you mean.
symtab.c is in CSRCS. (L40)
it triggers the test folder build. (with VPATH)
tests/symtab.c and $(FSIMG_HDR) are built with the same rule below, L62-63.

Comment (xiaoxiang781216) [reply examples/elf/Makefile]: Yes, if we let elf_main.c depends on tests/symtab.c, then make know that it should build symtab.c then test before elf_main.c, but if we change elf_main.c depends on $(FSIMG_HDR), but without any action, then how make ensure the header generated before it start to compile elf_main.c?

Comment (yamt) [reply examples/elf/Makefile]: * symtab.c and $(FSIMG_HDR) are built with the same rule (L62-63)
* elf_main.c (well, strictly speaking elf_main.o) depends on $(FSIMG_HDR) and dirlist.h, not symtab.c
* usually Make.dep will know elf_main.o depends on tests/dirlist.h (it was the reason for 45bcfa8b2eb1b0a84214a2bfc23f555ad8cf135e)

i don't understand why you prefer to make this depend on symtab.c over those headers.

Comment (xiaoxiang781216) [reply examples/elf/Makefile]: > * symtab.c and $(FSIMG_HDR) are built with the same rule (L62-63)

Yes.

> * elf_main.c (well, strictly speaking elf_main.o) depends on $(FSIMG_HDR) and dirlist.h, not symtab.c

Yes, that is why is it better to let elf_main.c depends on $(FSIMG_HDR) and dirlist.h.
 
> * usually Make.dep will know elf_main.o depends on tests/dirlist.h (it was the reason for [45bcfa8](https://github.com/apache/incubator-nuttx-apps/commit/45bcfa8b2eb1b0a84214a2bfc23f555ad8cf135e))
> 

Yes, but $(FSIMG_HDR) and dirlist.h is generated in another Makefile, there isn't these files in the first time invoke. If elf_main.c has no relationship with symtab.c, elf_main.c may compile first and than can't find these header file.

> i don't understand why you prefer to make this depend on symtab.c over those headers.

Comment (xiaoxiang781216) [reply examples/elf/Makefile]: @yamt we remove the first patch and use your solution, do you have any concern about this patch?

**Pull request merged**